src/NzbDrone.Core/MediaFiles/EpisodeImport/ImportApprovedEpisodes.cs

```diff
@@ -130,9 +149,19 @@ public List<ImportResult> Import(List<ImportDecision> decisions, bool newDownloa
                     episodeFile = _mediaFileService.Add(episodeFile);
                     importResults.Add(new ImportResult(importDecision));
 
+                    if (rename)
```

Comment:
I'm not sure I understand why this is necessary, the EpisodeFile isn't committed to the DB until line 149, any changes to episode file would then be committed, instead this waits for that to happen then immediately updates it.

Reply:
Why would a file that's just been imported need to immediately be renamed?

If the file name has changed that should be updated before committing, if it needs to be renamed after importing can we not calculate that before committing?

Reply:
Yes, that's broadly what I was thinking of wrt. "lower level methods"-using the basic "update media info, rename file" verbs that I am using currently relies on the file already being committed. But I'm guessing one can pull apart their functionality to actually do this to uncommitted files.